diff --git a/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_config.h b/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_config.h
--- a/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_config.h
+++ b/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_config.h
@@ -231,7 +231,7 @@
 /*
  * auto upgrade
  */
-#define FTS_AUTO_UPGRADE_EN                     1
+#define FTS_AUTO_UPGRADE_EN                     0
 
 /*
  * auto upgrade for lcd cfg
diff --git a/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_flash.c b/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_flash.c
--- a/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_flash.c
+++ b/drivers/input/touchscreen/mediatek/ft8719_spi/focaltech_flash.c
@@ -53,16 +53,9 @@
 * Global variable or extern global variabls/functions
 *****************************************************************************/
-u8 fw_file[] = {
-#include FTS_UPGRADE_FW_FILE
-};
-
-u8 fw_file2[] = {
-#include FTS_UPGRADE_FW2_FILE
-};
-
-u8 fw_file3[] = {
-#include FTS_UPGRADE_FW3_FILE
-};
+/* Xiaomi's OSS release omits the proprietary embedded touchscreen images. */
+u8 fw_file[] = { 0 };
+u8 fw_file2[] = { 0 };
+u8 fw_file3[] = { 0 };
 
 struct upgrade_module module_list[] = {
     {FTS_MODULE_ID, FTS_MODULE_NAME, fw_file, sizeof(fw_file)},
