diff --git a/drivers/media/platform/mtk-vcu/Makefile b/drivers/media/platform/mtk-vcu/Makefile
index c7a7a51..57bc7bd 100644
--- a/drivers/media/platform/mtk-vcu/Makefile
+++ b/drivers/media/platform/mtk-vcu/Makefile
@@ -4,7 +4,7 @@ ccflags-y += -I$(srctree)/drivers/misc/mediatek/cmdq
 mtk-vcu-y += mtk_vcu.o \
 	mtk_vcodec_mem.o \
 
-obj-$(CONFIG_VIDEO_MEDIATEK_VCU) += mtk-vcu.o mtk_vcodec_mem.o
+obj-$(CONFIG_VIDEO_MEDIATEK_VCU) += mtk-vcu.o
 ccflags-y += -I$(srctree)/arch/arm
 ccflags-y += -I$(srctree)/drivers/misc/mediatek/include/mt-plat
